src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java

```diff
@@ -1342,4 +1344,27 @@ void setDataSetBuilder(final DataSetBuilder dataSetBuilder) {
 	void setTrialEnvironmentBuilder(final TrialEnvironmentBuilder trialEnvironmentBuilder) {
 		this.trialEnvironmentBuilder = trialEnvironmentBuilder;
 	}
+
+	private boolean hasInvalidValues(List<FieldMapLabel> labels) {
+		if (!CollectionUtils.isEmpty(labels)) {
+			List<FieldMapLabel> invalid = labels.stream().filter(fieldMapLabel -> Util.getIntValue(fieldMapLabel.getColumn()) <= 0 || Util.getIntValue(fieldMapLabel.getRange()) <= 0).collect(Collectors.toList());
+			return !CollectionUtils.isEmpty(invalid);
+		}
+		return false;
+	}
+
+	private boolean hasOverlapping(List<FieldMapLabel> labels) {
```

Comment:
Rename to `hasOverlappingCoordinates`

Reply:
No need for `hasOverlapping` variable. Immediately return true once a duplicate is found. return false at the end (if no dupe found)